custom/ilsgateway/tanzania/handlers/soh.py

```diff
@@ -52,6 +52,7 @@ def _cleanup(s):
 class SohFormatter(Formatter):
 
     def format(self, text):
+        text = text.replace('\r', ' ').replace('\n', ' ').strip()
```

Comment:
Probably delivery handler also needs that kind of parsing. I will make another PR.